src/hotspot/share/jfr/jfr.hpp

```diff
@@ -60,9 +55,10 @@ class Jfr : AllStatic {
   static bool is_excluded(Thread* thread);
   static void include_thread(Thread* thread);
 
-#ifdef JFR_HAVE_INTRINSICS
+  // get_class_id intrinsic support
   static void trace_id_load_barrier(Klass* klass);
```

Comment:
I think we can make this general, because it is not necessarily specialized for the intrinsic use. Can we rename this to "get_class_id(const Klass* klass): instead? Then we can remove the comment // get_class_id intrinsic support., thanks

Reply:
Hmm, perhaps that was not such a good idea after all - it would require that you have tested for tagging at the call site. But "get_class_id_intrinsic(const Klass* klass):" might perhaps be ok.

Reply:
Good idea, updated.